test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -795,6 +795,53 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
 		framework.ExpectNoError(disableAutoprovisioning())
 	})
+
+	It("shouldn't add new node group if not needed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		By("Decrease size of groups")
+		nodesCountAfterResize := 0
+		for mig, size := range originalSizes {
+			if size > 1 {
+				framework.ExpectNoError(framework.ResizeGroup(mig, int32(1)))
+				nodesCountAfterResize++
+			}
+		}
+		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodesCountAfterResize, resizeTimeout))
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(enableAutoprovisioning(""))
+		By("Create pods")
+		ReserveMemory(f, "memory-reservation", nodesCountAfterResize+1, int(float64(nodesCountAfterResize+1)*float64(0.7)*float64(memAllocatableMb)), true, scaleUpTimeout)
+		defer framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.Namespace.Name, "memory-reservation")
+		By("Waiting for scale up")
+		// Verify that cluster size increased.
+		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
+			func(size int) bool { return size == nodesCountAfterResize+1 }, scaleUpTimeout))
+		By("Check if NAP group was created hoping id didn't happen")
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(disableAutoprovisioning())
+	})
+
+	It("shouldn't scale up if cores limit too low, should scale up after limit is changed [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+		Expect(getNAPNodePoolsNumber()).Should(Equal(0))
+		framework.ExpectNoError(enableAutoprovisioning("\"resource_limits\":{\"name\":\"cpu\", \"minimum\":2, \"maximum\":3}"))
```

Comment:
We should set resource limits based on cores already present in cluster.

Reply:
Changed.